### ISSUE TITLE
Moonlight Stuck On 'Quitting' Screen

### ISSUE THREAD

**Issue by atom0s** [moonlight-stream/moonlight-qt#415]
**Describe the bug**

Moonlight gets stuck closing a stream depending on how things are exited.

**Steps to reproduce**

 - Open a game via Moonlight. (ie. Path of Exile via Steam)
 - Once in-game, use the games menu to quit the game.
 - Moonlight will now show the hosts desktop. Press the quit shortcut: (CTRL+ALT+SHIFT+Q)
 - On the client, Path of Exile will still show as the session is running.
 - Click the 'X' on the game in the Moonlight window to kill that session.
 - Moonlight will now just sit at 'Quitting Path of Exile...' forever until force-closed.

**Screenshots**

https://i.imgur.com/iJnRTaL.png

**Affected games**

Seems to happen with any game I try and do the same steps as above.

**Other Moonlight clients**

N/A - I only use the Desktop streaming setup.

**Moonlight settings (please complete the following information)**

Here is a screenshot of my settings:
https://i.imgur.com/XMINHaf.png

**Client PC details (please complete the following information)**
 - OS: Windows 10 - 1607
 - Moonlight Version: v2.1.0

**Server PC details (please complete the following information)**
 - OS: Windows 10 - 1909
 - GeForce Experience version: v3.20.3.63
 - Nvidia GPU driver: v451.48

**Moonlight Logs (please attach)**

[Moonlight-1593288994.log](https://github.com/moonlight-stream/moonlight-qt/files/4841192/Moonlight-1593288994.log)
[Moonlight-1593287940.log](https://github.com/moonlight-stream/moonlight-qt/files/4841193/Moonlight-1593287940.log)

**Comment by cgutman** [comment]
It looks like Moonlight is just waiting for the cancel request forever, which indicates the real issue lies with GeForce Experience not responding in a timely manner. Does the game actually quit on the host?

We could handle this condition better though. I'll add a timeout to the cancel request.

**Comment by atom0s** [comment]
Yes. The game is closed manually by me/the user in this case. Moonlight is not the one closing it or requesting it be closed.

**Comment by cgutman** [comment]
Oh I see. So it's a manually added game, and therefore you have to quit the session yourself after. Then let me rephrase, is the running game indication on Path of Exile gone when you reopen Moonlight after it hangs like this?

**Comment by atom0s** [comment]
The game wasn't manually added. Moonlight sees it fine via Steam by default. I have not added any games to Moonlight manually.

When reopening after the hang, Moonlight does not see an active session still on the game.

**Comment by cgutman** [comment]
If it's an automatically detected game, the session should end when it closes. It could be launching a child process which remains running after the game is closed and GFE is mistakenly keeping the session alive for that.

If you just close the stream while the game is running, then use the quit app button, does it work then?
